lib/utils.js

```diff
@@ -17,3 +20,23 @@ exports.walk = function (dir, fs) {
 
   return results;
 };
+
+// remove a path recursively (async)
+function rmdir (fs, directory, callback) {
```

Comment:
I used this over fs-extra emptyDir because it is possible to set fs as a aoption and other fs implementation do not have those methods.

Reply:
@n1ru4l Do you think we should remove the `fs` option? I can't think of a reason to continue including it as an option. We make use of a few `fs-extra`-only methods anyway.

Reply:
I can not find any issue related to why this option exists. The only benefit I could think of is a in-memory-fs or something similar.

Maybe @jamhall can explain why it is there 😅

Reply:
I would rather see `fs-extra` used directly. Would make things much simpler to call `fsExtra.remove` than rolling out a homemade implementation.

I don't have an opinion on removing the `fs` option, but if we do we can cut a major release for it.

This is my only gripe with this PR – otherwise I want this merged so we can cut a minor release with this (and multipart uploads #78) soon. @n1ru4l let me know what you decide, we can always refactor in the next release.

Reply:
Let's keep it like this for now and release a new minor version and proceed further when @jamhall has answered.

Reply:
@leontastic Just realized the next release must be ma major release anyway since we dropped support for node versions prior to 6.